### ISSUE TITLE
memory consumption eruption for large golang file

### ISSUE THREAD

**Issue by mainred** [golangci/golangci-lint#530]
Thank you for creating the issue!

Please include the following information:
1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)
golangci-lint has version 1.15.0 built from 901cf25 on 2019-02-18T08:24:43Z
2. Config file: `cat .golangci.yml`
3. Go environment: `go version && go env`
go1.10.3
4. Verbose output of running: `golangci-lint run -v`

I happened to have a large golang file which is 28M, and golangci-lint is blocked in  `Go packages loading at mode load deps types and syntax`, for this sentence did not manged to print out, and golangci-lint is killed for OOM.
I found from `top` command that the memory usage spikes at terribly 9G.
```
20514 root      20   0 10.025g 9.831g      8 S  86.7 16.0   1:22.47 golangci-l+
```

**Comment by alexec** [comment]
See https://github.com/golangci/golangci-lint/issues/337

**Comment by tpounds** [comment]
Closing as duplicate of #337.

**Comment by jirfag** [comment]
Hi, I’m not sure it will be fixed in #337. It’s not a common use case. Why can’t you split file?
